### DISCUSSION TITLE
download large file with local mode

### DISCUSSION THREAD

**Discussion by LOVELYZOMBIEYHO** [python-telegram-bot/python-telegram-bot#3957] [Q&A]
Hi, I am struggling to build a Telegram bot for downloading large files (videos). First of all, all functions run successfully except when running locally. I have searched for how to run PTB on a local server, but it's not working. I have a few questions:

1. I understand that if I use a local server, the API should be [http://localhost:5002/bot{APIKEY}/getUpdates](http://localhost:5002/bot%7BAPIKEY%7D/getUpdates) or [http://127.0.0.1:5002/bot{APIKEY}/getUpdates](http://127.0.0.1:5002/bot%7BAPIKEY%7D/getUpdates) (using my port) rather than [https://api.telegram.org/bot{APIKEY}/getUpdates](https://api.telegram.org/bot%7BAPIKEY%7D/getUpdates). Is that correct?

2. How do I enable local mode and set it up? Some discussions mention that I can enable local server mode with the following code, but it's not working for me:
application=Application.builder().token(bot_token).local_mode(local_mode=True).base_file_url(BASE_FILE_URL).base_url(BASE_URL).build()
Is this the correct way to enable local mode?

3. Do you have an example of a local server setup? I found this one: [customwebhookbot.py](https://docs.python-telegram-bot.org/en/v20.6/examples.webappbot.html), but I'm not sure if it's what I need. If it is, is it necessary to set up a webhook? If not, could you please provide an example?

**Comment by Bibo-Joshi** [comment]
Hi. Please have a look at the wiki page https://github.com/python-telegram-bot/python-telegram-bot/wiki/Local-Bot-API-Server. I think that answers most of you questions.

**Reply by LOVELYZOMBIEYHO** [reply]
Sorry, I still don't know how to work with the local bot. After I read the wiki page, I think it works like this:

BASE_FILE_URL = 'http://localhost:5002/file/bot'
application = Application.builder().token(bot_token).local_mode(local_mode=True).base_file_url(BASE_FILE_URL).build()
But now, I don't know how to use or call the logOut function, and I'm not sure which one is correct: logOut() or log_out()?

I apologize for the beginner question. I also want to know if, when I want to upload/download large files, it's necessary to set local_mode=True. According to the documentation, it seems you only need to:

1. Host the files on your own server.
2. Use the logout function.
3. Change the base URL and base file URL.
Is that correct?"

**Reply by Bibo-Joshi** [reply]
* [logOut](https://docs.python-telegram-bot.org/en/v20.6/telegram.bot.html#telegram.Bot.logOut) is just an alias for [log_out](https://docs.python-telegram-bot.org/en/v20.6/telegram.bot.html#telegram.Bot.log_out), it doesn't matter which one you call.
* You can call `log_out` either in [`Application.post_init`](https://docs.python-telegram-bot.org/en/v20.6/telegram.ext.applicationbuilder.html#applicationbuilder) or complemetely independent of the `Application` do a one time call. How to make bot method calls without `Application` is described [here](https://github.com/python-telegram-bot/python-telegram-bot/wiki/Introduction-to-the-API). Ofc you can also log out by calling the endpoint directly in your browser.
* Yes, handling large files is supported by the bot API only with the `--local` flag and if you do that, you need to pass `local_mode=True` to PTB

**Reply by aiastia** [reply]
like this？


import asyncio
import telegram


async def main():
    bot = telegram.Bot("TOKEN")
    async with bot:
        print(await bot.logOut())


if __name__ == '__main__':
    asyncio.run(main())
